pkg/load/load.go

```diff
@@ -86,10 +86,11 @@ func configFromResolver(info *ResolverInfo) (*api.ReleaseBuildConfiguration, err
 
 // Registry takes the path to a registry config directory and returns the full set of references, chains,
 // and workflows that the registry's Resolver needs to resolve a user's MultiStageTestConfiguration
-func Registry(root string, flat bool) (references map[string]api.LiteralTestStep, chains map[string][]api.TestStep, workflows map[string]api.MultiStageTestConfiguration, err error) {
-	references = map[string]api.LiteralTestStep{}
-	chains = map[string][]api.TestStep{}
-	workflows = map[string]api.MultiStageTestConfiguration{}
+func Registry(root string, flat bool) (references registry.ReferenceMap, chains registry.ChainMap, workflows registry.WorkflowMap, documentation map[string]string, err error) {
+	references = registry.ReferenceMap{}
```

Comment:
nit: naming types after implementation specifics can also be fragile -- why not `registry.ReferencesByName`?